### PR TITLE
feat:  add omit function

### PR DESCRIPTION
**Pull request by nattallius**

### CODE REVIEW

**Comment by ala-n** [comment]
Discussed with a core team, the result proposed strategy is the following:
`copyDefinedKeys` is not going to be deprecated but should be defined as an alias  (PAF) for `omit` call with `(key, val) => val !== void 0` predicate.
'omit' is approved to be released in v2 stable release.

File: src/modules/esl-utils/misc/object.ts [hunk 1]
```diff
@@ -40,6 +43,16 @@ export function copyDefinedKeys<T>(obj?: T): Partial<T> {
   return result;
 }
 
+/** Makes a copy of obj with properties satisfying the predicate */
+export function omit(obj: any, predicate: (key: string, value?: any) => boolean): Record<string, any> {
```
Comment (ala-n) [inline src/modules/esl-utils/misc/object.ts:47]: Try to use template type instead see `copyDefinedKeys` definition above

Comment (ala-n) [inline src/modules/esl-utils/misc/object.ts:47]: There is also no need to define value: any as optional because it will be available anyway

**Review by YanaBr — APPROVED** [review]

**Review by andreybelous — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

File: src/modules/esl-utils/misc/object.ts [hunk 2]
```diff
@@ -33,13 +34,24 @@ export function defined<T>(...params: T[]) {
 
 /** Makes a flat copy without undefined keys */
 export function copyDefinedKeys<T>(obj?: T): Partial<T> {
+  return copy(obj || {}, (key, value) => value !== void 0);
+}
+
+/** Makes a plain copy of obj with properties satisfying the predicate
+ * If no predicate provided copies all own properties */
+export function copy<T>(obj: T, predicate: CopyPredicate  = () => true): Partial<T> {
```
Comment (ala-n) [inline src/modules/esl-utils/misc/object.ts:42]: Super-cosmetic: but it's better to place generic implementation fist, and the consumers (omit, copyDefKeys) after

File: src/modules/esl-utils/misc/object.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export const isObject = (obj: any): obj is Record<string, any> => obj && typeof obj === 'object';
 export const isObjectLike = (obj: any) => isObject(obj) || typeof obj === 'function';
+export type CopyPredicate = (key: string, value: any) => boolean;
```
Comment (ala-n) [inline src/modules/esl-utils/misc/object.ts:3]: Super-cosmetic: space to separate type exports from real objects

**Review by ala-n — COMMENTED** [review]
Also, update esl-utils README.md, please

PS: merge PR with squash to clearly describe the resulting change for changelog

File: src/modules/esl-utils/misc/object.ts [hunk 4]
```diff
@@ -31,15 +33,26 @@ export function defined<T>(...params: T[]) {
...
 }
 
+/** Makes a flat copy without undefined keys */
+export function copyDefinedKeys<T>(obj?: T): Partial<T> {
+  return copy(obj || {}, (key, value) => value !== void 0);
+}
+
+/** Omit copying provided properties from object */
+export function omit<T>(obj: T, keys: string[]) {
+  return copy(obj, key=> !keys.includes(key));
```
Comment (dshovchko) [inline src/modules/esl-utils/misc/object.ts:53]: typo: missing space

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-utils/misc/object.ts (hunk 4 above)
Comment (nattallius) [reply src/modules/esl-utils/misc/object.ts]: Thank you!

**Pull request merged**